src/secp256r1/curve.rs

```diff
@@ -76,19 +76,43 @@ new_curve_impl!(
     SECP_A,
     SECP_B,
     "secp256r1",
-    |curve_id, domain_prefix| sswu_hash_to_curve(curve_id, domain_prefix, Secp256r1::SSVDW_Z),
+    |domain_prefix| crate::hash_to_curve::hash_to_curve(domain_prefix, Secp256r1::default_hash_to_curve_suite()),
 );
 
 impl Secp256r1 {
     // Optimal Z with: <https://datatracker.ietf.org/doc/html/rfc9380#sswu-z-code>
     // 0xffffffff00000001000000000000000000000000fffffffffffffffffffffff5
     // Z = -10 (reference: <https://www.rfc-editor.org/rfc/rfc9380.html#section-8.2>)
-    const SSVDW_Z: Fp = Fp::from_raw([
+    const SSWU_Z: Fp = Fp::from_raw([
         0xfffffffffffffff5,
         0x00000000ffffffff,
         0x0000000000000000,
         0xffffffff00000001,
     ]);
+
+    fn default_hash_to_curve_suite() -> crate::hash_to_curve::Suite<Secp256r1, sha2::Sha256, 48> {
+        crate::hash_to_curve::Suite::<Secp256r1, sha2::Sha256, 48>::new(
+            b"P256_XMD:SHA-256_SSWU_RO_",
+            Self::SSWU_Z,
+            crate::hash_to_curve::Method::SSWU,
+        )
+    }
+}
+
+impl FromUniformBytes<48> for Fp {
```

Comment:
@davidnevadoc It would be messy for two reason:

1. It would be hard to generalize hash to curve suite for larger fields. For example pluto
2. For extension fields. For example `Fq2` of `bn256::G2` requires to implement `FromUniformBytes<96>` to use same hash to curve procedure

And totally agree with reusing `<64>`